pylint/config/callback_actions.py

```diff
@@ -38,3 +47,261 @@ def __call__(
+    """Display all the confidence levels that pylint knows about."""
+
+    def __call__(
+        self,
+        parser: argparse.ArgumentParser,
+        namespace: argparse.Namespace,
+        values: Union[str, Sequence[Any], None],
+        option_string: Optional[str] = "--list-conf-levels",
+    ) -> None:
+        for level in interfaces.CONFIDENCE_LEVELS:
+            print(f"%-18s: {level}")
+        sys.exit(0)
+
+
+class _ListExtensionsAction(_AccessRunObjectAction):
+    """Display all extensions under pylint.extensions."""
+
+    def __call__(
+        self,
+        parser: argparse.ArgumentParser,
+        namespace: argparse.Namespace,
+        values: Union[str, Sequence[Any], None],
+        option_string: Optional[str] = "--list-extensions",
+    ) -> None:
+        for filename in Path(extensions.__file__).parent.iterdir():
+            if filename.suffix == ".py" and not filename.stem.startswith("_"):
+                extension_name, _, _ = filename.stem.partition(".")
+                print(f"pylint.extensions.{extension_name}")
+        sys.exit(0)
+
+
+class _FullDocumentationAction(_AccessRunObjectAction):
+    """Display the full documentation."""
+
+    def __call__(
+        self,
+        parser: argparse.ArgumentParser,
+        namespace: argparse.Namespace,
+        values: Union[str, Sequence[Any], None],
+        option_string: Optional[str] = "--full-documentation",
+    ) -> None:
+        utils.print_full_documentation(self.run.linter)
+        sys.exit(0)
+
+
+class _GenerateRCFileAction(_AccessRunObjectAction):
+    """Generate a pylintrc file."""
+
+    def __call__(
+        self,
+        parser: argparse.ArgumentParser,
+        namespace: argparse.Namespace,
+        values: Union[str, Sequence[Any], None],
+        option_string: Optional[str] = "--generate-rcfile",
+    ) -> None:
+        self.run.linter.generate_config(skipsections=("COMMANDS",))
+        sys.exit(0)
+
+
+class _ErrorsOnlyModeAction(_AccessRunObjectAction):
+    """Turn on errors-only mode.
+
+    Error mode:
+        * disable all but error messages
+        * disable the 'miscellaneous' checker which can be safely deactivated in
+          debug
+        * disable reports
+        * do not save execution information
+    """
+
+    def __call__(
+        self,
+        parser: argparse.ArgumentParser,
+        namespace: argparse.Namespace,
+        values: Union[str, Sequence[Any], None],
+        option_string: Optional[str] = "--errors-only",
+    ) -> None:
+        self.run.linter.error_mode()
+
+
+class _VerboseModeAction(_AccessRunObjectAction):
+    """Turn on verbose mode."""
+
+    def __call__(
+        self,
+        parser: argparse.ArgumentParser,
+        namespace: argparse.Namespace,
+        values: Union[str, Sequence[Any], None],
+        option_string: Optional[str] = "--verbose",
+    ) -> None:
+        self.run.verbose = True
+
+
+class _EnableAllExtensionsAction(_AccessRunObjectAction):
```

Comment:
We could add a tests for this Action as it's untested right now.

Reply:
We can't since this is done via the `preprocessing` thing in `Run`. We should probably do that via these callbacks as well, but let's do that in another PR.

Reply:
Same goes for `--verbose` by the way.